web/forms/implementing_simple_autocompletion/my_element.html

```diff
@@ -1,26 +1,28 @@
-<link rel="import" href="match_element.html">
-<link rel="import" href="packages/core_elements/core_input.html">
-<polymer-element name="my-element">
+<dom-module id="my-element">
   <template>
     <style>
       ul.candidate-list li {
         list-style-type: none;
       }
+
       ul.candidate-list li .selected {
         background-color: #ddd;
       }
     </style>
-    <input is="core-input" on-keyup="{{findCandidates}}" on-keydown="{{keyDown}}" placeholder="Type slowly">
-    <template if="{{!matchingCandidates.isEmpty}}">
+    <input is="paper-input" on-keyup="findCandidates" on-keydown="keyDown"
+           placeholder="Type slowly">
+    <template is="dom-if" if="{{matchingCandidates.length}}">
       <ul class="candidate-list">
-        <template repeat="{{candidate in matchingCandidates}}">
+        <template is="dom-repeat" items="{{matchingCandidates}}"
+                  as="candidate">
           <li>
-            <match-element class="{{ { 'selected': candidate == selectedCandidate } }}"
-                           value="[[candidate]]"
-                           inputQuery="[[inputQuery]]"></match-element>
+            <match-element
+                class$="{{selectedClass(candidate)}}"
```

Comment:
add `selectedCandidate` as a 2nd argument here, so it gets updated if that changes:

`class$="{{selectedClass(candidate, selectedCandidate)}}"`